src/Robo/Commands/Setup/BuildCommand.php

```diff
@@ -134,4 +134,45 @@ public function installAlias() {
       ->run();
   }
 
+  /**
+   * Checks the repo if there is modified files.
+   */
+  public function checkCleanRepo() {
+    $clean = $this->taskExec("git status --porcelain")
+      ->dir($this->getConfigValue('repo.root'))
+      ->interactive()
+      ->setVerbosityThreshold(VerbosityThresholdInterface::VERBOSITY_VERBOSE)
+      ->run();
+    return $clean;
+  }
+
+  /**
+   * Adds a project dependency and ensures configuration management is in sync.
+   *
+   * @command add-dependency
+   */
+  public function addDependency($dependency) {
+    $modified = $this->checkCleanRepo();
+    if (!empty($modified)) {
+      $this->output()->writeln("<comment>Please note the following files need to be commited or reverted:</comment>");
+      $this->output()->writeln("<comment>$modified</comment>");
+    } else {
+      $this->taskExec("composer update $dependency --with-dependencies")
```

Comment:
So, this method isn't actually updating a dependency. It's updating one, so maybe it should be renamed.